user_sessions/middleware.py

```diff
@@ -1,68 +1,17 @@
-import time
-
 from django.conf import settings
-from django.utils.cache import patch_vary_headers
-from django.utils.http import http_date
-
-try:
-    from importlib import import_module
-except ImportError:
-    from django.utils.importlib import import_module
+from django.contrib.sessions.middleware import (
+    SessionMiddleware as DjangoSessionMiddleware,
+)
 
-try:
-    from django.utils.deprecation import MiddlewareMixin
-except ImportError:
-    class MiddlewareMixin:
-        pass
 
-
-class SessionMiddleware(MiddlewareMixin):
+class SessionMiddleware(DjangoSessionMiddleware):
     """
     Middleware that provides ip and user_agent to the session store.
     """
     def process_request(self, request):
-        engine = import_module(settings.SESSION_ENGINE)
         session_key = request.COOKIES.get(settings.SESSION_COOKIE_NAME, None)
-        request.session = engine.SessionStore(
+        request.session = self.SessionStore(
```

Comment:
Won't this be an instance of django's `SessionStore` instead of ours?

Reply:
In a properly configured project, no.

Now that we subclass Django's middleware, it will import `settings.SESSION_ENGINE` and assign things properly. See [here](https://github.com/django/django/blob/7bb741d787ba360a9f0d490db92e22e0d28204ed/django/contrib/sessions/middleware.py#L13).

Reply:
Oh, the attribute name is unfortunately confusing (should be `self.session_store`), but that's an upstream issue.

Reply:
GitHub really needs a 💯 reaction emoji.

But regardless, consider your comment here 💯'd.